build.zig

```diff
@@ -292,10 +293,12 @@ pub fn build(b: *Builder) !void {
                 artifact.addCSourceFiles(&optimized_c_sources, &[_][]const u8{ "-std=c99", "-O3" });
 
                 artifact.linkLibrary(softfloat);
-                artifact.linkLibCpp();
+                if (artifact.target.getOs().tag != .windows or artifact.target.getAbi() != .msvc) {
+                    artifact.linkLibCpp();
```

Comment:
Why don't we want to link against `libc++` when targeting `.msvc` ABI? Is this circumventing Zig wanting to use its own-built `libc++.a` when `linkLibCpp` is specified?

Reply:
This was to handle this problem: https://github.com/ziglang/zig/issues/12703#issuecomment-1304604393

I tried just not building/linking `libcxxabi` at first, which gets you as far as linking (diffs I tried in the comment there). 

The problem is that libc++ once built/linked under the msvc-built zig, conflicts with libcmt.lib (which comes from the msvc-built llvm that links with it):

```
lld-link : error : duplicate symbol: void * __cdecl operator new(unsigned __int64) [C:\cygwin64\home\kcbanner\kit\zig\build-release\stage3.vcxproj]
  >>> defined at d:\a01\_work\12\s\src\vctools\crt\vcstartup\src\heap\new_scalar.cpp:32
  >>>            libcmt.lib(new_scalar.obj)
  >>> defined at C:\cygwin64\home\kcbanner\kit\zig\lib\libcxx\src/new.cpp
  >>>            libc++.lib(new.obj)
```

Reply:
Actually, I think I was too hasty here. This might only be necessary during bootstrap - when we're linking against an msvc-built llvm (and not a zig-built one). 

I'm going to attempt bootstrapping with this check in place, then removing it when building self-hosted. It might be that the solution is a build arg like `-Duse-native-libcxx` to gate this behavior might be the real solution.

Reply:
Updated this flow to use a new `-Ddisable-libcpp` flag passed in by the bootstrapping script when building on a `*-windows-msvc` host.